pkg/agent/openflow/pipeline_windows.go

```diff
@@ -281,5 +281,41 @@ func (c *client) hostBridgeUplinkFlows(localSubnet net.IPNet, category cookie.Ca
 			Cookie(c.cookieAllocator.Request(category).Raw()).
 			Done(),
 	}
+	if c.encapMode.SupportsNoEncap() {
+		// If NoEncap is enabled, the reply packets from remote Pod can be forwarded to local Pod directly.
+		// by explicitly resubmitting them to endpointDNATTable and marking "macRewriteMark" at same time.
+		flows = append(flows, c.pipeline[conntrackStateTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+			MatchRegRange(int(marksReg), markTrafficFromUplink, binding.Range{0, 15}).
+			MatchDstIPNet(localSubnet).
+			Action().LoadRegRange(int(marksReg), macRewriteMark, macRewriteMarkRange).
+			Action().GotoTable(endpointDNATTable).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Done())
+	}
 	return flows
 }
+
+func (c *client) l3FwdFlowToRemoteViaRouting(localGatewayMAC net.HardwareAddr, remoteGatewayMAC net.HardwareAddr,
+	category cookie.Category, peerIP net.IP, peerPodCIDR *net.IPNet) []binding.Flow {
+	if !c.encapMode.NeedsRoutingToPeer(peerIP, c.nodeConfig.NodeIPAddr) && remoteGatewayMAC != nil {
+		// It enhances Windows Noencap mode performance by bypassing host network.
+		ipProto := getIPProtocol(peerPodCIDR.IP)
+		l3FwdTable := c.pipeline[l3ForwardingTable]
+		var flows []binding.Flow
+		flows = append(flows, l3FwdTable.BuildFlow(priorityNormal).MatchProtocol(ipProto).
```

Comment:
Isn't the flow equal to `c.l3FwdFlowToRemoteViaGW(remoteGatewayMAC, *peerPodCIDR, category)`?

Reply:
Yes, we can reuse `l3FwdFlowToRemoteViaGW()`.